tests/canary_upgrade_test.go

```diff
@@ -79,17 +80,15 @@ var _ = Describe("[Serial][sig-operator]virt-handler canary upgrade", Serial, de
 
 	AfterEach(func() {
 		close(stopCh)
+		patchPayload, err := patch.New(patch.WithReplace("/spec/customizeComponents", originalKV.Spec.CustomizeComponents)).GeneratePayload()
+		Expect(err).ToNot(HaveOccurred())
+		_, err = virtCli.KubeVirt(flags.KubeVirtInstallNamespace).Patch(context.Background(), originalKV.Name, types.JSONPatchType, patchPayload, metav1.PatchOptions{})
+		Expect(err).ToNot(HaveOccurred())
 
-		retry.RetryOnConflict(retry.DefaultRetry, func() error {
-			_, err := virtCli.KubeVirt(flags.KubeVirtInstallNamespace).Update(context.Background(), originalKV, metav1.UpdateOptions{})
-			return err
-		})
-
-		Eventually(func() bool {
-			ds, err := virtCli.AppsV1().DaemonSets(originalKV.Namespace).Get(context.Background(), "virt-handler", metav1.GetOptions{})
-			Expect(err).ToNot(HaveOccurred())
-			return ds.Status.DesiredNumberScheduled == ds.Status.NumberReady && ds.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable.IntValue() == 1
-		}, 60*time.Second, 1*time.Second).Should(BeTrue(), "waiting for virt-handler to be ready")
+		ds, err := virtCli.AppsV1().DaemonSets(flags.KubeVirtInstallNamespace).Get(context.Background(), "virt-handler", metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(ds.Status.DesiredNumberScheduled).To(Equal(ds.Status.NumberReady))
+		Expect(ds.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable.IntValue()).To(Equal(1))
```

Comment:
The `Eventually` block here should stay since it is not immediate that the status is propagated to the virt-handler.

Reply:
Thanks, i figured as much, but i wasn't sure..